Editorial: More changes for comparisons

jmdyck opened the pull request:
... in line with PR #2877 and PR #3202.

The first commit changes:
  `_X_ and _Y_ are [not] the same {Realm,Module} Record`
to just:
  `_X_ is [not] _Y_`

The second commit changes:
  `SameValue(_X_, _Y_) is {*true*,*false*}`
to
  `_X_ is [not] _Y_`
when the two values are both Strings, or else one is a String and the other is a String or `*null*`.

jmdyck commented:
> ... I think one of the other editors preferred to keep "the same X Record" when we talked about this last

It seemed odd to me that #2877 didn't eliminate that form, so I checked the PR's comments for any discussion of it, but didn't find anything.

bakkot commented:
That sounds like the sort of opinion I'd have. I do kind of prefer "the same X Record" because it makes it really, really clear that comparison is by identity rather than structural, without you having to keep in mind the details of how identity works for each kind of spec value.

jmdyck commented:
> it makes it really, really clear that comparison is by identity rather than structural,

So are there other kinds of values for which you'd like this clarity?

bakkot commented:
Only really Records and Lists.

bakkot commented:
Per call, we mildly prefer to keep the "same Record as" form, extending I guess to using it in the existing `Repeat, while _thisEnv_ is not _varEnv_` case.

bakkot commented:
But we do like the change to use `is` instead of `SameValue` for comparing strings.

jmdyck commented:
Okay, I've dropped the first commit (reinstating the status quo uses of `are the same Foo Record`), and I've added some more commits to align the spec with comparison conventions.

jmdyck commented:
(force-pushed to squash it down to one commit)
